src/data_access/FirebaseRoomsDataAccessObject.java

```diff
@@ -248,4 +248,22 @@ public void deleteRoom(User user, LoginUserDataAccessInterface userDAO, Room roo
       throw new RuntimeException("Unable to delete room. Please try again.");
     }
   }
+
+  @Override
+  public void changeRoomName(
+      User user, LoginUserDataAccessInterface userDAO, Room activeRoom, String roomName) {
+    String idToken = userDAO.getAccessToken(user.getEmail(), user.getPassword());
+    String jsonBody = JSONObject.quote(roomName);
+    String url = String.format(Constants.ROOM_NAME_URL, activeRoom.getUid()) + "?auth=" + idToken;
+    RequestBody body = RequestBody.create(jsonBody, MediaType.parse("application/json"));
+    Request request = new Request.Builder().url(url).put(body).build();
+    try {
+      Response response = client.newCall(request).execute();
+      if (!response.isSuccessful()) {
+        throw new IOException();
+      }
+    } catch (IOException | JSONException e) {
+      throw new RuntimeException("Unable to change room name. Please try again.");
```

Comment:
You should be passing an error given from the call to Firebase. You can choose to fix this error later on in the interactor or presenter.

Reply:
This is exactly what all the other methods in this file do... What exactly do you want here?

Reply:
Include error messages from Firebase

Reply:
You should throw something similar to `RuntimeException(response.message)`, feed this to the interactor, and then the view should display something like "Unable to do ___" + the response error message from Firebase